src/nemos/basis/_basis_mixin.py

```diff
@@ -0,0 +1,260 @@
+
+from ..convolve import create_convolutional_predictor
+from ._transformer_basis import TransformerBasis
+
+
+class EvalBasisMixin:
+    """Mixin class for evaluational basis."""
+
+    def __init__(self, bounds: Optional[Tuple[float, float]] = None):
+        self.bounds = bounds
+
+    def _compute_features(self, *xi: ArrayLike):
+        """
+        Apply the basis transformation to the input data.
+
+        The basis evaluated at the samples, or :math:`b_i(*xi)`, where :math:`b_i` is a
+        basis element. xi[k] must be a one-dimensional array or a pynapple Tsd.
+
+        Parameters
+        ----------
+        *xi:
+            The input samples over which to apply the basis transformation. The samples can be passed
+            as multiple arguments, each representing a different dimension for multivariate inputs.
+
+        Returns
+        -------
+        :
+            A matrix with the transformed features. The basis evaluated at the samples,
+            or :math:`b_i(*xi)`, where :math:`b_i` is a basis element. xi[k] must be a one-dimensional array
+            or a pynapple Tsd.
+
+        """
+        return self._evaluate(*xi)
+
+    def _set_kernel(self) -> "EvalBasisMixin":
+        """
+        Prepare or compute the convolutional kernel for the basis functions.
+
+        For EvalBasisMixin, this method might not perform any operation but simply return the
+        instance itself, as no kernel preparation is necessary.
+
+        Returns
+        -------
+        self :
+            The instance itself.
+
+        """
+        return self
+
+    @property
+    def bounds(self):
+        """Range of values covered by the basis."""
+        return self._bounds
+
+    @bounds.setter
+    def bounds(self, values: Union[None, Tuple[float, float]]):
+        """Setter for bounds."""
+        if values is not None and len(values) != 2:
+            raise ValueError(
+                f"The provided `bounds` must be of length two. Length {len(values)} provided instead!"
+            )
+
+        # convert to float and store
+        try:
+            self._bounds = values if values is None else tuple(map(float, values))
+        except (ValueError, TypeError):
+            raise TypeError("Could not convert `bounds` to float.")
+
+        if values is not None and values[1] <= values[0]:
+            raise ValueError(
+                f"Invalid bound {values}. Lower bound is greater or equal than the upper bound."
+            )
+
+
+class ConvBasisMixin:
+    """Mixin class for convolutional basis."""
+
+    def __init__(self, window_size: int, conv_kwargs: Optional[dict] = None):
+        self.window_size = window_size
+        self.conv_kwargs = {} if conv_kwargs is None else conv_kwargs
+
+    def _compute_features(self, *xi: ArrayLike):
+        """
+        Apply the basis transformation to the input data.
+
+        A bank of basis filters (created by calling fit) is convolved with the
```

Comment:
this breaks

> This docstring needs to be reworked. in particular:


>     * Top-line should be "Convolves basis functions with inputs" or similar

That's a good point but it will break my test logic  (i was using this first sentence to check the content of the "compute_features", I can have a regex that matches multiple strings

>     * Remove reference to fit
ok
>     * That big paragraph needs to be broken up. I think the sentences that start "For example" should at least be new paragraphs, if not demonstrated as part of the examples.
> 
>     * This docstring mixes "samples" and "input data" to refer to the input
I tried to use input consistently
>     * it says input can be ndarray or pynapple, but the type-hinting just says arraylike. is that right?

good call, it is actually a FeatureMatrix (array like would work too but it is never passed before conversion to array, since it is called by `compute_features`)

Reply:
I went ahead and made a couple more changes here, including removing the reference to fit and dtypes. Since we're not describing anything else about dtype, no need to specify it here.